api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -518,19 +534,28 @@ public void setConditions(Set<Condition> conditions) {
 	 * @param condition - the condition to add
 	 */
 	public void addCondition(Condition condition) {
-		condition.setEncounter(this);
-		getConditions().add(condition);
+		if (conditions == null) {
+			conditions = new LinkedHashSet<>();
+		}
+
+		if (condition != null) {
+			condition.setEncounter(this);
+			conditions.add(condition);
+		}
 	}
-	
+
 	/**
 	 * Remove the given condition from the set of conditions for this Encounter
 	 *
 	 * @param condition - the condition to remove
 	 */
 	public void removeCondition(Condition condition) {
-		if (conditions != null) {
-			conditions.remove(condition);
-		}
+		
+		Optional.ofNullable(conditions).orElse(new LinkedHashSet<>()).stream().filter(c -> !c.getVoided() && c.getUuid().equals(condition.getUuid())).forEach(c -> {
+			c.setVoided(true);
+			c.setDateVoided(new Date());
+			c.setVoidedBy(Context.getAuthenticatedUser());
```

Comment:
We should have a void reason when voiding the condition, even if it's as simple as "Voided by the API"

Reply:
Done.

Reply:
@ibacher I didn't find any good example in Core, so we went for the message that you suggested.